packages/gatsby/content/advanced/editor-sdks.md

```diff
@@ -65,6 +65,49 @@ yarn pnpify --sdk
 
 3. Set [`tsserver.tsdk`](https://github.com/neoclide/coc-tsserver#configuration-options) to `.vscode/pnpify/typescript/lib`
 
+### Emacs
+
+The SDK comes with a typescript-language-server wrapper which enables you to use the ts-ls LSP client.
+
+1. Add PnPify to your dependencies:
+
+```bash
+yarn add @yarnpkg/pnpify
+```
+
+2. Run the following command, which will generate a new directory called `.vscode/pnpify`:
+
+```bash
+yarn pnpify --sdk
+```
+
+3. Create a `.dir-locals.el` with the following content to enable Flycheck and LSP support:
+
+```lisp
+((typescript-mode
+  . (
+     ;; Enable typescript-language-server and eslint LSP clients.
+     (lsp-enabled-clients . (ts-ls eslint))
+     (eval . (let ((project-directory (car (dir-locals-find-file "."))))
+               (set (make-local-variable 'flycheck-javascript-eslint-executable)
+                    (concat project-directory ".vscode/pnpify/eslint/bin/eslint.js"))
+
+               (lsp-dependency 'typescript-language-server
+                               `(:system ,(concat project-directory ".vscode/pnpify/typescript-language-server/lib/cli.js")))
+               (lsp-dependency 'typescript
+                               `(:system ,(concat project-directory ".vscode/pnpify/typescript/bin/tsserver")))
+
+               ;; Re-(start) LSP to pick up the dependency changes above.
+               (lsp)
+               )))))
+```
+
+4. You may have to install `vscode-jsonrpc` and `vscode-languageserver-protocol` manually if `typescript-language-server` fails to start (they are not listed in its dependencies as of this writing):
```

Comment:
Would it make sense to add those dependencies to the [builtin `packageExtensions`](https://github.com/yarnpkg/berry/blob/master/packages/plugin-compat/sources/extensions.ts)?

Reply:
Certainly! I'm pretty sure the experience becomes less confusing.